superset/databases/api.py

```diff
@@ -238,6 +238,7 @@ def post(self) -> Response:
         except DatabaseInvalidError as ex:
             return self.response_422(message=ex.normalized_messages())
         except DatabaseConnectionFailedError as ex:
+            logger.warning("Database connection failed: %s", item["sqlalchemy_uri"])
```

Comment:
This log will be masked uri

Reply:
It appears that `item["sqlalchemy_uri"` is only updated with the masked version if line 236 is reached - if `CreateDatabaseCommand(...).run()` fails, as we can expect if this code is reached, then I believe there's a good chance that the full `sqlalchemy_uri` will be logged.

While the masked URI is better than the full URI, it still exposes username and additional connection information that it might be better not to have in logs.